runtime-extension-model/src/main/java/org/mule/runtime/config/internal/validation/ExpressionsInRequiredExpressionsParams.java

```diff
@@ -57,7 +56,7 @@ public Level getLevel() {
     // According to the extension model, no collections or target-value for foreach, parallel-foreach, etc...
     // must be defined by an expression, but this was not enforced. This check is needed to avoid breaking on
     // legacy cases
-    return ERROR;
+    return WARN;
```

Comment:
see comment in https://github.com/mulesoft/mule-integration-tests/pull/1646#pullrequestreview-838798399